main.cpp

```diff
@@ -1 +1,238 @@
+#include <iostream>
+#include <string>
+#include <cstdlib>
+#include <ctime>
+#include <vector>
+#include <charconv>
+#include <algorithm>
+#include <iterator>
+#include <stdio.h>
+
+using namespace std;
+
+void print(int guesses = 0) { // print the hangman
+if (guesses >= 1) {cout << "  | ";}
+if (guesses >= 2) {cout << "\n  | ";}
+if (guesses >= 3) {cout << "\n  O ";}
+if (guesses == 4) {cout << "\n /";}
+if (guesses == 5) {cout << "\n /|";}
+if (guesses >= 6) {cout << "\n /|\\";}
+if (guesses == 7) {cout << "\n / ";}
+if (guesses >= 8) {cout << "\n / \\";}
+if (guesses >= 9) {cout << "\n ___";}
+if (guesses == 10) {cout << "\n|   |\nYOU LOST!";}
+}
+
+void printoptions() { //print options
+    cout << "\nSelect One The Following Option:" <<
+            "\n[1] Guess a letter" <<
+            "\n[2] Guess the word" <<
+            "\n[3] Check guessed letters" <<
+            "\n[4] View options again" <<
+            "\n[5] Quit game" << endl <<
+            "[6] Check Current Word (FOR TESTING ONLY)\n";
+}
+
+void printword(string word, vector<char> cg) { // function to print
+    int i;
+    for (i = 0; i < word.length(); ++i) {
+        if (count(cg.begin(), cg.end(), word[i])) {
+        std::cout << word[i] << " ";
+    }
+    else
+    cout << "_ ";
+    }
+
+}
+
+
+int main(){
+    char input{}; //for user's letter guesses
+    int choice{}; //for user choice
+    string wordinput{}; //for when user wants to guess the entire word
+    vector<char> guessedchar{}; //array for all the letters the user have guessed
+    vector<string> list{"special", "random", "roast", "firetruck", "throw", "trend", "blast", "gasoline"};
+    vector<char> correctguesses{}; //letters that were guessed correctly
+    int guesses{}; // number of guesses by the user
+    
+    srand (time(NULL));
+    int random = rand() % list.size(); //randomise the word
+    string word = list[random];
+    
+    
+    printoptions();
+
+    while (guesses <= 10) { 
+        cout << "Remaining Guesses: " << 10 - guesses << endl;
+        print(guesses); //print hangman
+        cout << endl;
+
+        if (guesses == 10) { //if user has lost the game
```

Comment:
You have used the hardcoded value '10' 3 times already. If you want to change the max number of guesses, you have to change it in 3 places. Turn it into a const variable, or used define